src/span_context.js

```diff
@@ -140,7 +123,19 @@ export default class SpanContext {
   }
 
   set flags(flags: number): void {
-    this._flags = flags;
+    this._samplingState.setFlags(flags);
+  }
+
+  setIsSampled(value: boolean) {
```

Comment:
Let's try to keep most new methods private

Reply:
SGTM.